datumaro/plugins/icdar_format/converter.py

```diff
@@ -0,0 +1,211 @@
+
+
+class _WordRecognitionConverter():
+    def __init__(self):
+        self.annotations = ''
+
+    def save_categories(self, save_dir, categories):
+        pass
+
+    def save_annotations(self, item, categories):
+        self.annotations += '%s, ' % (item.id + IcdarPath.IMAGE_EXT)
+        for ann in item.annotations:
+            if ann.type != AnnotationType.caption:
+                continue
+            self.annotations += '\"%s\"' % ann.caption
+        self.annotations += '\n'
+
+    def write(self, path):
+        file = osp.join(path, 'gt.txt')
+        os.makedirs(osp.dirname(file), exist_ok=True)
+        with open(file, 'w') as f:
+            f.write(self.annotations)
+
+    def is_empty(self):
+        return len(self.annotations) == 0
+
+class _TextLocalizationConverter():
+    def __init__(self):
+        self.annotations = {}
+
+    def save_categories(self, save_dir, categories):
+        vocabulary_file = osp.join(save_dir,
+            IcdarPath.TASK_DIR[IcdarTask.text_localization],
+            IcdarPath.VOCABULARY_FILE)
+        os.makedirs(osp.dirname(vocabulary_file), exist_ok=True)
+        with open(vocabulary_file, 'w', encoding='utf-8') as f:
+            f.write('\n'.join(l.name
+                for l in categories[AnnotationType.label])
+            )
+
+    def save_annotations(self, item, categories):
+        annotation = ''
+        for ann in item.annotations:
+            if ann.type == AnnotationType.bbox:
+                annotation += '%s %s %s %s' % (ann.x, ann.y,
+                    ann.x + ann.w, ann.y + ann.h)
+                if ann.label is not None:
+                    annotation += ' %s' % \
+                        categories[AnnotationType.label][ann.label].name
+            elif ann.type == AnnotationType.polygon:
+                annotation += ','.join(str(p) for p in ann.points)
+                if ann.label is not None:
+                    annotation += ',\"%s\"' % \
+                        categories[AnnotationType.label][ann.label].name
+            annotation += '\n'
+        self.annotations[item.id] = annotation
+
+    def write(self, path):
+        os.makedirs(path, exist_ok=True)
+        for item in self.annotations:
+            file = osp.join(path, 'gt_' + item + '.txt')
+            with open(file, 'w') as f:
+                f.write(self.annotations[item])
+
+    def is_empty(self):
+        return len(self.annotations) == 0
+
+class _TextSegmentationConverter():
+    def __init__(self):
+        self.annotations = {}
+        self.masks = {}
+
+    def save_categories(self, save_dir, categories):
+        pass
+
+    def save_annotations(self, item, categories):
+        annotation = ''
+        colormap = [(255, 255, 255)]
+        group = 0
+        for ann in item.annotations:
+            if ann.type == AnnotationType.mask:
+                char = ''
+                if ann.group != group:
+                    annotation += '\n'
+                    group += 1
```

Comment:
I think, there are no guarantees the annotations will be sorted by group number. Take a look at `utils.annotation_utils.find_instances` for reference.